src/vmm_lwt.ml

```diff
@@ -12,13 +12,13 @@ let safe_close fd =
     (fun () -> Lwt_unix.close fd)
     (fun _ -> Lwt.return_unit)
 
-let port_socket port =
+let port_socket ip port =
   let open Lwt_unix in
   let s = socket PF_INET6 SOCK_STREAM 0 in
   set_close_on_exec s ;
   setsockopt s SO_REUSEADDR true ;
   setsockopt s IPV6_ONLY false ;
-  bind s (ADDR_INET (Unix.inet6_addr_any, port)) >>= fun () ->
+  bind s (ADDR_INET (Ipaddr_unix.to_inet_addr ip, port)) >>= fun () ->
```

Comment:
Hmm, I worry if we specify `0.0.0.0` that we get the same error on Linux.

Reply:
Yes, we do.

Reply:
uhm, so should we change some code? we could use above `PF_INET` if we have an ipv4 address?

Reply:
I think we can do a dance by converting to ipv6 with `Ipaddr.(V6 (to_v6 ip))` as I just suggested below.

Reply:
👍🏾

Reply:
I just tested binding using the above code with "0.0.0.0", and it doesn't fail, but I don't see it listening in the output from `netstat -tlnp`. I will try to test some more...

Reply:
This article suggests that there may be issues with using ::ffff:0.0.0.0 on some platforms: https://lore.kernel.org/all/87tz5s1pxl.fsf@nemi.mork.no/T/